Core/Object Arts/Dolphin/MVP/Views/Common Controls/ListView.cls

```diff
@@ -1474,41 +1474,51 @@ moreParamsAt: n put: anObject
-	((itemClicked isNil and: [anyKeysDown])
-		or: [event button ~= #left and: [anyKeysDown or: [newSelections includes: itemClicked]]])
-			ifTrue: [^newSelections].
-	""
+	"If the click is outside all items, and ctrl or shift is down, nothing changes."
+	(itemClicked isNil and: [anyKeysDown]) ifTrue: [^newSelections].
+	"A right or middle click on an already-selected item also does not change the selection,
+	otherwise it would be very frustrating to open the context menu with multiple items selected.
+	The control also chooses to leave the selection alone if modifiers are held during a right/middle click."
+	(event button ~= #left and: [anyKeysDown or: [newSelections includes: itemClicked]])
+		ifTrue: [^newSelections].
+	"Having taken care of the above edge cases, if no keys are down, the behavior is equivalent to the single-select case."
 	anyKeysDown ifFalse: [^itemClicked ifNil: [#()] ifNotNil: [:item | Array with: item]].
-	"Only Ctrl"
+	"Only Ctrl is down, so we flip the selection state of the clicked item."
 	event isShiftDown ifFalse: 
 			[newSelections := newSelections asSortedCollection.
 			newSelections remove: itemClicked ifAbsent: [newSelections add: itemClicked].
 			^newSelections asArray].
-	"At least Shift, start dealing with the selection mark"
+	"At least Shift is down, so we must start examining the 'selection mark'. This is the last item
+	that was clicked with no modifiers, only ctrl, or ctrl + shift (shift-clicks do *not* move the
+	selection mark). It acts as the 'start point' for operations where shift is held--they affect
+	the range of items from the selection mark to the click point, inclusive. Initially it is zero,
+	i.e. out-of-bounds, in which case holding shift essentially has no effect, and we act only on
+	the item actually clicked."
 	markIndex := self selectionMarkIndex.
 	markRange := markIndex == 0
 				ifTrue: [itemClicked to: itemClicked]
 				ifFalse: [(markIndex min: itemClicked) to: (markIndex max: itemClicked)].
-	"Only Shift"
+	"Only shift is down, not ctrl, so the current selection will be entirely replaced by the marked range."
 	event isCtrlDown ifFalse: [^markRange asArray].
-	"Ctrl and Shift"
+	"Both ctrl and shift are down, so we set the selection state of the marked range to that of the
+	selection mark itself. Note that this ignores the selection state of the item actually clicked
+	and everything in between it and the selection mark."
 	newSelections := newSelections asSet.
 	markIsSelected := newSelections includes: markIndex.
 	markRange do: (markIsSelected
 				ifTrue: [[:i | newSelections add: i]]
 				ifFalse: [[:i | newSelections remove: i ifAbsent: []]]).
-	"For some reason, the item clicked is always left selected,
-	even if we are deselecting the range (markIsSelected == false).
-	This is not true in e.g. Windows Explorer--why?"
+	"For some reason, the item clicked is always left selected, even if we are deselecting the rest
```

Comment:
Hmm...any idea where I might look to find the cause? At a glance, I'm not seeing anywhere that interacts with the selection while it is changing other than `onButtonPressed:` itself.